omnipaxos/src/sequence_paxos/leader.rs

```diff
@@ -217,9 +217,11 @@ where
             .leader_state
             .get_decided_idx(*pid)
             .expect("Received PromiseMetaData but not found in ld");
-        let (delta_snapshot, suffix, sync_idx) =
-            if followers_decided_idx < my_decided_idx && T::Snapshot::use_snapshots() {
-                // Send missing decided entries in snapshot and accepted entries in suffix
+        let (delta_snapshot, suffix, sync_idx) = if followers_promise_n == max_promise_n {
+            // Follower's accepted entries are valid, send any new entries after
+            if T::Snapshot::use_snapshots() && *followers_accepted_idx < my_decided_idx {
+                // Note: we snapshot from follower's decided and not follower's accepted because
+                // snapshots currently can't handle merging onto accepted entries.
```

Comment:
We don't do the most optimal thing here which would be to send the follower a snapshot of entries (follower_acc_idx, leader_dec_idx) since the followers accepted entries are actually decided the follower just doesn't know it yet. However we would have to send some flag telling the follower to merge onto its accepted entries instead of onto its decided entries (`SnapshotType::Delta(Snapshot, bool)` or something). It makes the snapshot code messier than it already is.

Snapshotting like this is only meaningfully better than the current snapshotting when the follower has received a bunch of accepts from the previous leader but not their corresponding decides. With how sequence numbers work this shouldn't really happen.